metals/src/main/scala/scala/meta/internal/pantsbuild/BloopPants.scala

```diff
@@ -354,156 +336,72 @@ private class BloopPants(
-    def id: String = value.obj("id").str
-    def isTestTarget: Boolean = targetType == "TEST"
-    def isResourceTarget: Boolean = targetType == "RESOURCE"
-  }
-  private def isSupportedTargetType(target: Value): Boolean =
-    // Instead of whitelisting which target types we support, we hardcode which
-    // known target types we know we don't support. Pants plugins can introduce
-    // new target types that are minor variations of for example `scala_library`
-    // that we may want to support.
-    !unsupportedTargetType.contains(target.pantsTargetType)
-
-  private def toBloopProject(
-      id: String,
-      target: Value
-  ): C.Project = {
-
-    val baseDirectories = for {
-      roots <- target.obj.get("roots").toList
-      root <- roots.arr
-      sourceRoot <- root.obj.get("source_root")
-    } yield Paths.get(sourceRoot.str)
-
-    val baseDirectory = id.baseDirectory
-
-    val sources =
-      if (target.isResourceTarget) Nil
-      else filemap.forTarget(id).toList
-
-    // Extract target dependencies.
-    val dependsOn = (for {
-      dependency <- target.obj("targets").arr
-      acyclicDependency = cycles.acyclicDependency(dependency.str)
-      if acyclicDependency != id
-      if isSupportedTargetType(targets(acyclicDependency))
-    } yield acyclicDependency).toList
-
-    // Extract class directories of direct target dependencies.
-    val dependencyClassdirectories: List[Path] = (for {
-      dependency <- target.obj("targets").arr
-      classDir <- makeClassdirectory(dependency.str)
-    } yield classDir).toList
-
-    // Extract 3rd party library dependencies, and their associated sources.
-    val libraryDependencies = for {
-      dependency <- target.obj("libraries").arr
-      library <- libraries.get(dependency.str)
+  private def toBloopProject(target: PantsTarget): C.Project = {
+
+    val baseDirectory: Path = PantsConfiguration
+      .baseDirectory(AbsolutePath(workspace), target.name)
+      .toNIO
+
+    val sources: List[Path] =
+      if (target.targetType.isResource) Nil
+      else filemap.forTarget(target.name).toList
+
+    val transitiveDependencies: List[PantsTarget] = (for {
+      dependency <- target.transitiveDependencies
+      if dependency != target.name
+    } yield export.targets(dependency)).toList
+
+    val dependencies: List[String] = for {
+      dependency <- transitiveDependencies
+      // Rewrite dependencies on targets that belong to a cyclic component.
+      acyclicDependencyName = cycles.acyclicDependency(dependency.name)
+      if acyclicDependencyName != target.name
+      acyclicDependency = export.targets(acyclicDependencyName)
+      if acyclicDependency.isTargetRoot && !acyclicDependency.targetType.isAnyResource
+    } yield acyclicDependency.name
+
+    val libraries: List[PantsLibrary] = for {
+      dependency <- transitiveDependencies
+      libraryName <- dependency.libraries
+      library <- export.libraries.get(libraryName).toList
```

Comment:
nit: I think the `.toList` is unnecessary

Reply:
done